backends/apple/mps/runtime/operations/IndexingOps.mm

```diff
@@ -206,25 +206,30 @@
 
 Error
 MPSGraphBuilder::mpsScatterOp(NodePtr nodePtr) {
-  auto graphNode = nodePtr->mpsnode_union_as_MPSScatter();
-  ET_LOG(
-    Debug, "%s %d: %d",
-    __FUNCTION__, graphNode->input1_id(), graphNode->output_id()
-  );
+  if (@available(iOS 16, macOS 13, *)) {
+    auto graphNode = nodePtr->mpsnode_union_as_MPSScatter();
+    ET_LOG(
+      Debug, "%s %d: %d",
+      __FUNCTION__, graphNode->input1_id(), graphNode->output_id()
+    );
 
-  int64_t dim = graphNode->dim();
-  MPSGraphTensor* inputTensor = getMPSGraphTensor(graphNode->input1_id());
-  MPSGraphTensor* indicesTensor = getMPSGraphTensor(graphNode->idx_id());
-  MPSGraphTensor* updatesTensor = getMPSGraphTensor(graphNode->src_id());
+    int64_t dim = graphNode->dim();
+    MPSGraphTensor* inputTensor = getMPSGraphTensor(graphNode->input1_id());
+    MPSGraphTensor* indicesTensor = getMPSGraphTensor(graphNode->idx_id());
+    MPSGraphTensor* updatesTensor = getMPSGraphTensor(graphNode->src_id());
 
-  _idToMPSGraphTensor[graphNode->output_id()] =
-    [_mpsGraph scatterAlongAxis:dim
-                 withDataTensor:inputTensor
-                  updatesTensor:updatesTensor
-                  indicesTensor:indicesTensor
-                           mode:MPSGraphScatterModeSet
-                           name:nil];
-  return Error::Ok;
+    _idToMPSGraphTensor[graphNode->output_id()] =
+      [_mpsGraph scatterAlongAxis:dim
+                   withDataTensor:inputTensor
+                    updatesTensor:updatesTensor
+                    indicesTensor:indicesTensor
+                             mode:MPSGraphScatterModeSet
+                             name:nil];
+
+    return Error::Ok;
+  } else {
+    return Error::NotSupported;
```

Comment:
Thanks, one minor thing - can you add an error log here like
```
ET_LOG(Error, "...")
```

Reply:
Added, and I also changed the requirements to iOS 15.4 and MacOS 12.3